File: 2302-count-subarrays-with-score-less-than-k/2302-count-subarrays-with-score-less-than-k.py
class Solution:
    def countSubarrays(self, nums: List[int], k: int) -> int:
        total = 0
        window = 0
        left = 0
        count = 0

        for right, value in enumerate(nums):
            total += value
            window += 1
        
            while total * window >= k:
                total -= nums[left]
                window -= 1
                left += 1
            count += window
        return count

File: 2302-count-subarrays-with-score-less-than-k/2302-count-subarrays-with-score-less-than-k.java
class Solution {
    public long countSubarrays(int[] nums, long k) {
        long count = 0;
        long total = 0;
        
        int left = 0;
        int window = 0;

        for (int right = 0; right < nums.length; right++) {
            total += nums[right];
            window++;

            while ((total * window) >= k) {
                total -= nums[left];
                window--;
                left++;
            }
            count += window;
        }

        return count;
    }
}